docs/proposals/pod-security-context.md

```diff
@@ -0,0 +1,233 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<strong>
+The latest 1.0.x release of this document can be found
+[here](http://releases.k8s.io/release-1.0/docs/proposals/apiserver_watch.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+## Abstract
+
+A proposal for refactoring `SecurityContext` to have pod-level and container-level attributes in
+order to correctly model pod- and container-level security concerns.
+
+## Motivation
+
+Currently, containers have a `SecurityContext` attribute which contains information about the
+security settings the container uses.  In practice many of these attributes are uniform across all
+containers in a pod.  Simultaneously, there is also a need to apply the security context pattern
+at the pod level to correctly model security attributes that apply only at a pod level.
+
+Users should be able to:
+
+1.  Express security settings that are applicable to the entire pod
+2.  Express base security settings that apply to all containers
+3.  Override only the settings that need to be differentiated from the base in individual
+    containers
+
+Goals of this design:
+
+1.  Describe the use cases for which a pod-level security context is necessary
+2.  Describe the necessary model changes and associated implementation changes
+
+## Constraints and assumptions
+
+1.  We will not design for intra-pod security; we are not currently concerned about isolating
+    containers in the same pod from one another
+1.  We will design for backward compatibility with the current V1 API
+
+## Use Cases
+
+1.  As a developer, I want to correctly model security attributes which belong to an entire pod
```

Comment:
Hrm, I'm not sure users necessarily care about how the pod-level security API is factored, as long as they have the knobs and the knobs work correctly.